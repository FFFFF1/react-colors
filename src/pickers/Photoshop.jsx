import React from 'react'
import PsPointer from './PsPointer.jsx'
import ColorPicker, {
	ColorInput, 
	ColorRadio, 
	ColorPanel, 
	ColorBar
} from '../components'


class Photoshop extends React.Component {
	static defaultProps = {
	  color: [255,0,0]
	}

	constructor(props) {
	  super(props)
	
	  this.state = {
	  	color: props.color,
	  	currColor: props.color
	  }
	}

	handleChange = (newColor) => {
		console.log(newColor)
		this.setState({color: newColor})
	}

	handleConfirm = () => {
		this.setState({currColor: this.state.color})
	}

	render () {
		let {color, currColor} = this.state

		const styles = {
			root: {
				width: 567,
				height: 354,
				boxShadow: 'rgba(0, 0, 0, 0.247059) 0px 0px 0px 1px, rgba(0, 0, 0, 0.14902) 0px 8px 16px',
				background: '#ededed'
			},
			title: {
				boxSizing: 'border-box',
				position: 'absolute',
				color: '#666',
				left: 0,
				top: 0,
				width: '100%',
				height: 26,
				lineHeight: '26px',
				paddingLeft: 4, 
				fontSize: 13,
				background: '#fff'
			},
			panel: {
				position: 'absolute',
				left: 12,
				top: 57,
				width: 255,
				height: 255
			},
			bar: {
				display: 'absolute',
				left: 280,
				top: 57,
				width: 20,
				height: 255
			},
			color: {
				position: 'absolute',
				left: 315,
				top: 44,
				width: 60,
				height: 68,
			},
			colorP: {
				margin: 0,
				fontSize: 13,
				textAlign: 'center'
			},
			buttons: {
				position: 'absolute',
				left: 436,
				top: 36
			},
			colorNew: {
				height: 34,
				background: `rgb(${Math.round(color[0])},${Math.round(color[1])},${Math.round(color[2])})`
			},
			colorCurr: {
				height: 34,
				background: `rgb(${Math.round(currColor[0])},${Math.round(currColor[1])},${Math.round(currColor[2])})`
			},
			inputs: {
				position: 'absolute',
				left: 315,
				top: 167,
			},
			input: {
				marginBottom: 4,
				width: 100,
				height: 19
			} 
		}

		return (
			<ColorPicker style={ styles.root } defaultColor={this.props.color} colorModel='rgb' onChange={ (a) => this.handleChange(a) }>	
				<div style={ styles.title }>Color Picker</div>
				<ColorPanel style={ styles.panel }/>
				<ColorBar pointer={ <PsPointer /> } style={ styles.bar }/>
				<div style={ styles.color }>
					<p style={ styles.colorP }>new</p>
					<div style={ styles.colorNew }></div>
					<div style={ styles.colorCurr }></div>
					<p style={ styles.colorP }>current</p>
				</div>
				<div style={ styles.buttons }>
					<button onClick={ this.handleConfirm }>OK</button>
				</div>
				<div style={ styles.inputs } data-color='1'>
					<div data-color='1' style={ styles.input }><ColorRadio model='hsv.h' /><ColorInput label='H:' model='hsv.h'/></div>
					<div data-color='1' style={ styles.input }><ColorRadio model='hsv.s' /><ColorInput label='S:' model='hsv.s'/></div>
					<div data-color='1' style={ styles.input }><ColorRadio model='hsv.v' /><ColorInput label='V:' model='hsv.v'/></div>
					<div style={{height: 4}}></div>
					<div data-color='1' style={ styles.input }><ColorRadio model='rgb.r' /><ColorInput label='R:' model='rgb.r'/></div>
					<div data-color='1' style={ styles.input }><ColorRadio model='rgb.g' /><ColorInput label='G:' model='rgb.g'/></div>
					<div data-color='1' style={ styles.input }><ColorRadio model='rgb.b' /><ColorInput label='B:' model='rgb.b'/></div>
				</div>
			</ColorPicker>
		)
	}
}

export default Photoshop