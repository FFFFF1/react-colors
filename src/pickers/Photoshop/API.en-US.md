| Property       | Description           | Type             | Default       |
|------------|----------------|------------------|--------------|
| color      | color | one of string or array | [255,0,0,1]  |
| colorModel | color input model | string |  -  |
| onChange   | color change callback function  | function    | - |
| onAccept   | click accept button callback function  | function    | - |
| onCancel  | click cancel callback function  | function    | - |
| title 		| title | node | 'Color Picker' |

