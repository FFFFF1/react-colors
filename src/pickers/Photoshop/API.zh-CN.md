| Property       | Description           | Type             | Default       |
|------------|----------------|------------------|--------------|
| color      | 颜色值 | one of string or array | [255,0,0,1]  |
| colorModel | 输入颜色的色彩空间 | string |  -  |
| onChange   | 颜色值变化回调  | function    | - |
| onAccept   | 点击确认按钮回调  | function    | - |
| onCancel  | 点击取消按钮回调  | function    | - |
| title 		| 头部 | node | 'Color Picker' |
