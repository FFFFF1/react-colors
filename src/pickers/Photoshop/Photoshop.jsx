import React, {PropTypes} from 'react'
import chroma from 'chroma-js'
import PsPointerDoubleTriangle from './PsPointerDoubleTriangle'
import PsPointerCircle from './PsPointerCircle'
import ColorPicker, {
	ColorInput, 
	ColorRadio, 
	ColorPanel, 
	ColorBar,
	ColorBlock
} from '../../components'
import getHexColor from '../../helpers/rgbToHex'

class Photoshop extends React.Component {
	static propTypes = {
		color: PropTypes.oneOfType([PropTypes.array, PropTypes.string]),
		title: PropTypes.node,
		onChange: PropTypes.func,
		onAccept: PropTypes.func,
		onCancel: PropTypes.func,
		model: PropTypes.string,
		colorModel: PropTypes.string
	}

	static defaultProps = {
		title: 'Color Picker',
	  color: [255,0,0,1],
	  onChange: () => {},
	  onAccept: () => {},
	  onCancel: () => {}
	}

	constructor(props) {
	  super(props)
		let color = {rgba: chroma(props.color, props.colorModel).rgba()}
	  this.state = {
	  	color: color,
	  	currColor: color
	  }
	}

	componentWillReceiveProps(nextProps) {
		let color = {rgba: chroma(nextProps.color, nextProps.colorModel).rgba()}
		this.setState({color: color})
	}

	handleChange = (newColor) => {
		this.setState({color: newColor})
		this.props.onChange(newColor)
	}

	handleAccept = () => {
		this.setState({currColor: this.state.color})
		this.props.onAccept(this.state.color)
	}

	handleCancel = () => {
		this.props.onCancel()
	}

	render () {
		let {color, currColor} = this.state

		return (
			<ColorPicker style={ styles.root } color={color.rgba} colorModel='rgba' onChange={ (a) => this.handleChange(a) }>	
				<div style={ styles.title }>{ this.props.title }</div>
				<ColorPanel pointer={ <PsPointerCircle /> } style={ styles.panel }/>
				<ColorBar direction='vertical' pointer={ <PsPointerDoubleTriangle /> } style={ styles.bar }/>
				<div style={ styles.color } data-color='1'>
					<p style={ styles.colorP }>new</p>
					<ColorBlock color={ getHexColor(color.rgba) } style={ styles.colorBlock }></ColorBlock>
					<ColorBlock color={ getHexColor(currColor.rgba) } style={ styles.colorBlock }></ColorBlock>
					<p style={ styles.colorP }>current</p>
				</div>
				<div style={ styles.buttons }>
					<button style={ styles.button } onClick={ this.handleAccept }>OK</button>
					<button style={ styles.button } onClick={ this.handleCancel }>Cancel</button>
				</div>
				<div style={ styles.inputHsv } data-color='1'>
					<div data-color='1' style={styles.inputDiv}><ColorRadio model='hsv.h'/><ColorInput inputStyle={styles.input} label='H:' model='hsv.h' rightLabel='Â°'/></div>
					<div data-color='1' style={styles.inputDiv}><ColorRadio model='hsv.s'/><ColorInput inputStyle={styles.input} label='S:' model='hsv.s' rightLabel='%' scale={100}/></div>
					<div data-color='1' style={styles.inputDiv}><ColorRadio model='hsv.v'/><ColorInput inputStyle={styles.input} label='V:' model='hsv.v' rightLabel='%' scale={100}/></div>
				</div>
				<div style={ styles.inputRgb } data-color='1'>
					<div data-color='1' style={styles.inputDiv}><ColorRadio model='rgb.r'/><ColorInput inputStyle={styles.input} label='R:' model='rgb.r'/></div>
					<div data-color='1' style={styles.inputDiv}><ColorRadio model='rgb.g'/><ColorInput inputStyle={styles.input} label='G:' model='rgb.g'/></div>
					<div data-color='1' style={styles.inputDiv}><ColorRadio model='rgb.b'/><ColorInput inputStyle={styles.input} label='B:' model='rgb.b'/></div>
				</div>
				<div style={ styles.inputLab } data-color='1'>
					<div data-color='1' style={styles.inputDiv}><ColorInput inputStyle={styles.lab} label='L:' model='lab.l'/></div>
					<div data-color='1' style={styles.inputDiv}><ColorInput inputStyle={styles.lab} label='a:' model='lab.a'/></div>
					<div data-color='1' style={styles.inputDiv}><ColorInput inputStyle={styles.lab} label='b:' model='lab.b'/></div>
				</div>
				<div style={ styles.inputCmyk } data-color='1'>
					<div data-color='1' style={styles.inputDiv}><ColorInput inputStyle={styles.input} label='C:' model='cmyk.c' rightLabel='%' scale={100}/></div>
					<div data-color='1' style={styles.inputDiv}><ColorInput inputStyle={styles.input} label='M:' model='cmyk.m' rightLabel='%' scale={100}/></div>
					<div data-color='1' style={styles.inputDiv}><ColorInput inputStyle={styles.input} label='Y:' model='cmyk.y' rightLabel='%' scale={100}/></div>
					<div data-color='1' style={styles.inputDiv}><ColorInput inputStyle={styles.input} label='K:' model='cmyk.k' rightLabel='%' scale={100}/></div>
				</div>
					<div style={ styles.inputHex } data-color='1'>
					<ColorInput inputStyle={ styles.hex } model='hex'/>
				</div>
				
			</ColorPicker>
		)
	}
}

const styles = {
	root: {
		width: 550,
		height: 354,
		boxShadow: 'rgba(0, 0, 0, 0.247059) 0px 0px 0px 1px, rgba(0, 0, 0, 0.14902) 0px 8px 16px',
		background: '#ededed',
		fontSize: 'Robot nono Sans-serif'
	},
	title: {
		boxSizing: 'border-box',
		position: 'absolute',
		color: '#666',
		left: 0,
		top: 0,
		width: '100%',
		height: 26,
		lineHeight: '26px',
		paddingLeft: 4, 
		fontSize: 13,
		background: '#fff'
	},
	panel: {
		position: 'absolute',
		left: 12,
		top: 57,
		width: 255,
		height: 255
	},
	bar: {
		display: 'absolute',
		left: 280,
		top: 57,
		width: 20,
		height: 255
	},
	color: {
		position: 'absolute',
		left: 315,
		top: 44,
		width: 60,
		height: 68,
	},
	colorP: {
		margin: 0,
		fontSize: 13,
		textAlign: 'center'
	},
	buttons: {
		position: 'absolute',
		left: 426,
		top: 36
	},
	button: {
		width: 115,
		height: 20,
		marginBottom: 10 
	},
	colorBlock: {
		height: 34,
	},
	inputHsv: {
		position: 'absolute',
		left: 315,
		top: 167,
	},
	inputRgb: {
		position: 'absolute',
		left: 315,
		top: 244,
	},
	inputLab: {
		position: 'absolute',
		left: 455,
		top: 167
	},
	inputCmyk: {
		position: 'absolute',
		left: 455,
		top: 244,
	},
	inputHex: {
		position: 'absolute',
		left: 318,
		top: 319,
	},
	inputDiv: {
		marginBottom: 4,
		height: 19
	},
	input: {
		width: 45,
		height: 20
	},
	lab: {
		width: 56,
		height: 20
	},
	hex: {
		width: 80,
		height: 20
	} 
	}

export default Photoshop