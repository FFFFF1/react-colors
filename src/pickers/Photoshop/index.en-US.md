look like color picker in Adobe Photoshop.