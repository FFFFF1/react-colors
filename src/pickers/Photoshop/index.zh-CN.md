模仿 Adobe Photoshop 中的拾色器