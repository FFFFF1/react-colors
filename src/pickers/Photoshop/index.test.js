import React from 'react'
import Photoshop from './Photoshop.jsx'
import { mount, render } from 'enzyme'
import { renderToJson } from 'enzyme-to-json'

describe('ColorHexInput', () => {
	it('render correctly', () => {
		const wrapper = render(<Photoshop color='#2196F3'/>)
		expect(renderToJson(wrapper)).toMatchSnapshot()
	})
})