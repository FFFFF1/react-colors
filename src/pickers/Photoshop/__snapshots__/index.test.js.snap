// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorHexInput render correctly 1`] = `
<div
  style="position:relative;width:550px;height:354px;box-shadow:rgba(0, 0, 0, 0.247059) 0px 0px 0px 1px, rgba(0, 0, 0, 0.14902) 0px 8px 16px;background:#ededed;font-size:Robot nono Sans-serif;"
>
  <div
    style="box-sizing:border-box;position:absolute;color:#666;left:0;top:0;width:100%;height:26px;line-height:26px;padding-left:4px;font-size:13px;background:#fff;"
  >
    Color Picker
  </div>
  <div
    style="position:absolute;height:255px;background:linear-gradient(to top, #000, transparent),linear-gradient(to right, #FFF, rgba(255,255,255,0)),
            linear-gradient(to top, hsl(206.57142857142858, 100%, 50%), hsl(206.57142857142858, 100%, 50%));cursor:default;left:12px;top:57px;width:255px;"
  >
    <div
      style="position:absolute;top:4.705882352941182%;left:86.41975308641975%;"
    >
      <div
        style="width:10px;height:10px;box-shadow:0 0 0 1.5px #fff, inset 0 0 1px 1px rgba(0,0,0,.3),
            0 0 1px 2px rgba(0,0,0,.4);border-radius:50%;transform:translate(-5px, -5px);"
      />
    </div>
  </div>
  <div
    style="position:relative;height:255px;background-image:linear-gradient(to top, #f00 0%, #ff0 17%, #0f0 33%,
            #0ff 50%, #00f 67%, #f0f 83%, #f00 100%);curosr:default;display:absolute;left:280px;top:57px;width:20px;"
  >
    <div
      style="position:absolute;top:42.61904761904761%;"
    >
      <div
        style="position:absolute;width:40px;transform:translate(-10px, -5px);"
      >
        <svg
          fill="#fff"
          height="10px"
          stroke="#666"
          style="float:left;transform:rotate(180deg);"
          version="1.1"
          width="10px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 5 L6 1 L8 1 A2 2 0 0 1 9 2 L9 8 A2 2 0 0 1 8 9 L6 9 L0 5"
          />
        </svg>
        <svg
          fill="#fff"
          height="10px"
          stroke="#666"
          style="float:right;"
          version="1.1"
          width="10px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 5 L6 1 L8 1 A2 2 0 0 1 9 2 L9 8 A2 2 0 0 1 8 9 L6 9 L0 5"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    data-color="1"
    style="position:absolute;left:315px;top:44px;width:60px;height:68px;"
  >
    <p
      style="margin:0;font-size:13px;text-align:center;"
    >
      new
    </p>
    <div
      style="height:34px;background-image:linear-gradient(rgb(33,150,243),rgb(33,150,243),rgb(33,150,243),rgb(33,150,243), rgb(33,150,243),rgb(33,150,243),rgb(33,150,243),rgb(33,150,243)),
        									linear-gradient(45deg, #ccc 25%, transparent 25%, transparent 75%, #ccc 75%, #ccc),
        									linear-gradient(45deg, #ccc 25%, transparent 25%, transparent 75%, #ccc 75%, #ccc);background-size:100% 100%,12px 12px,12px 12px;background-position:0 0,0 0, 6px 6px;background-color:#fff;"
    />
    <div
      style="height:34px;background-image:linear-gradient(rgb(33,150,243),rgb(33,150,243),rgb(33,150,243),rgb(33,150,243), rgb(33,150,243),rgb(33,150,243),rgb(33,150,243),rgb(33,150,243)),
        									linear-gradient(45deg, #ccc 25%, transparent 25%, transparent 75%, #ccc 75%, #ccc),
        									linear-gradient(45deg, #ccc 25%, transparent 25%, transparent 75%, #ccc 75%, #ccc);background-size:100% 100%,12px 12px,12px 12px;background-position:0 0,0 0, 6px 6px;background-color:#fff;"
    />
    <p
      style="margin:0;font-size:13px;text-align:center;"
    >
      current
    </p>
  </div>
  <div
    style="position:absolute;left:426px;top:36px;"
  >
    <button
      style="width:115px;height:20px;margin-bottom:10px;"
    >
      OK
    </button>
    <button
      style="width:115px;height:20px;margin-bottom:10px;"
    >
      Cancel
    </button>
  </div>
  <div
    data-color="1"
    style="position:absolute;left:315px;top:167px;"
  >
    <div
      data-color="1"
      style="margin-bottom:4px;height:19px;"
    >
      <input
        checked=""
        type="radio"
      />
      <div
        style="display:inline-block;position:relative;margin-bottom:4px;"
      >
        <span
          style="display:inline-block;height:100%;font-size:13px;vertical-align:top;width:20px;line-height:20px;"
        >
          H:
        </span>
        <input
          style="vertical-align:top;width:45px;box-sizing:border-box;height:20px;padding:0 2px;font-size:12px;"
          type="text"
          value="207"
        />
        <span
          style="display:inline-block;height:100%;font-size:13px;vertical-align:top;padding-left:4px;line-height:20px;"
        >
          Â°
        </span>
      </div>
    </div>
    <div
      data-color="1"
      style="margin-bottom:4px;height:19px;"
    >
      <input
        type="radio"
      />
      <div
        style="display:inline-block;position:relative;margin-bottom:4px;"
      >
        <span
          style="display:inline-block;height:100%;font-size:13px;vertical-align:top;width:20px;line-height:20px;"
        >
          S:
        </span>
        <input
          style="vertical-align:top;width:45px;box-sizing:border-box;height:20px;padding:0 2px;font-size:12px;"
          type="text"
          value="86"
        />
        <span
          style="display:inline-block;height:100%;font-size:13px;vertical-align:top;padding-left:4px;line-height:20px;"
        >
          %
        </span>
      </div>
    </div>
    <div
      data-color="1"
      style="margin-bottom:4px;height:19px;"
    >
      <input
        type="radio"
      />
      <div
        style="display:inline-block;position:relative;margin-bottom:4px;"
      >
        <span
          style="display:inline-block;height:100%;font-size:13px;vertical-align:top;width:20px;line-height:20px;"
        >
          V:
        </span>
        <input
          style="vertical-align:top;width:45px;box-sizing:border-box;height:20px;padding:0 2px;font-size:12px;"
          type="text"
          value="95"
        />
        <span
          style="display:inline-block;height:100%;font-size:13px;vertical-align:top;padding-left:4px;line-height:20px;"
        >
          %
        </span>
      </div>
    </div>
  </div>
  <div
    data-color="1"
    style="position:absolute;left:315px;top:244px;"
  >
    <div
      data-color="1"
      style="margin-bottom:4px;height:19px;"
    >
      <input
        type="radio"
      />
      <div
        style="display:inline-block;position:relative;margin-bottom:4px;"
      >
        <span
          style="display:inline-block;height:100%;font-size:13px;vertical-align:top;width:20px;line-height:20px;"
        >
          R:
        </span>
        <input
          style="vertical-align:top;width:45px;box-sizing:border-box;height:20px;padding:0 2px;font-size:12px;"
          type="text"
          value="33"
        />
        <span
          style="display:inline-block;height:100%;font-size:13px;vertical-align:top;padding-left:4px;line-height:20px;"
        />
      </div>
    </div>
    <div
      data-color="1"
      style="margin-bottom:4px;height:19px;"
    >
      <input
        type="radio"
      />
      <div
        style="display:inline-block;position:relative;margin-bottom:4px;"
      >
        <span
          style="display:inline-block;height:100%;font-size:13px;vertical-align:top;width:20px;line-height:20px;"
        >
          G:
        </span>
        <input
          style="vertical-align:top;width:45px;box-sizing:border-box;height:20px;padding:0 2px;font-size:12px;"
          type="text"
          value="150"
        />
        <span
          style="display:inline-block;height:100%;font-size:13px;vertical-align:top;padding-left:4px;line-height:20px;"
        />
      </div>
    </div>
    <div
      data-color="1"
      style="margin-bottom:4px;height:19px;"
    >
      <input
        type="radio"
      />
      <div
        style="display:inline-block;position:relative;margin-bottom:4px;"
      >
        <span
          style="display:inline-block;height:100%;font-size:13px;vertical-align:top;width:20px;line-height:20px;"
        >
          B:
        </span>
        <input
          style="vertical-align:top;width:45px;box-sizing:border-box;height:20px;padding:0 2px;font-size:12px;"
          type="text"
          value="243"
        />
        <span
          style="display:inline-block;height:100%;font-size:13px;vertical-align:top;padding-left:4px;line-height:20px;"
        />
      </div>
    </div>
  </div>
  <div
    data-color="1"
    style="position:absolute;left:455px;top:167px;"
  >
    <div
      data-color="1"
      style="margin-bottom:4px;height:19px;"
    >
      <div
        style="display:inline-block;position:relative;margin-bottom:4px;"
      >
        <span
          style="display:inline-block;height:100%;font-size:13px;vertical-align:top;width:20px;line-height:20px;"
        >
          L:
        </span>
        <input
          style="vertical-align:top;width:56px;box-sizing:border-box;height:20px;padding:0 2px;font-size:12px;"
          type="text"
          value="60"
        />
        <span
          style="display:inline-block;height:100%;font-size:13px;vertical-align:top;padding-left:4px;line-height:20px;"
        />
      </div>
    </div>
    <div
      data-color="1"
      style="margin-bottom:4px;height:19px;"
    >
      <div
        style="display:inline-block;position:relative;margin-bottom:4px;"
      >
        <span
          style="display:inline-block;height:100%;font-size:13px;vertical-align:top;width:20px;line-height:20px;"
        >
          a:
        </span>
        <input
          style="vertical-align:top;width:56px;box-sizing:border-box;height:20px;padding:0 2px;font-size:12px;"
          type="text"
          value="2"
        />
        <span
          style="display:inline-block;height:100%;font-size:13px;vertical-align:top;padding-left:4px;line-height:20px;"
        />
      </div>
    </div>
    <div
      data-color="1"
      style="margin-bottom:4px;height:19px;"
    >
      <div
        style="display:inline-block;position:relative;margin-bottom:4px;"
      >
        <span
          style="display:inline-block;height:100%;font-size:13px;vertical-align:top;width:20px;line-height:20px;"
        >
          b:
        </span>
        <input
          style="vertical-align:top;width:56px;box-sizing:border-box;height:20px;padding:0 2px;font-size:12px;"
          type="text"
          value="-55"
        />
        <span
          style="display:inline-block;height:100%;font-size:13px;vertical-align:top;padding-left:4px;line-height:20px;"
        />
      </div>
    </div>
  </div>
  <div
    data-color="1"
    style="position:absolute;left:455px;top:244px;"
  >
    <div
      data-color="1"
      style="margin-bottom:4px;height:19px;"
    >
      <div
        style="display:inline-block;position:relative;margin-bottom:4px;"
      >
        <span
          style="display:inline-block;height:100%;font-size:13px;vertical-align:top;width:20px;line-height:20px;"
        >
          C:
        </span>
        <input
          style="vertical-align:top;width:45px;box-sizing:border-box;height:20px;padding:0 2px;font-size:12px;"
          type="text"
          value="86"
        />
        <span
          style="display:inline-block;height:100%;font-size:13px;vertical-align:top;padding-left:4px;line-height:20px;"
        >
          %
        </span>
      </div>
    </div>
    <div
      data-color="1"
      style="margin-bottom:4px;height:19px;"
    >
      <div
        style="display:inline-block;position:relative;margin-bottom:4px;"
      >
        <span
          style="display:inline-block;height:100%;font-size:13px;vertical-align:top;width:20px;line-height:20px;"
        >
          M:
        </span>
        <input
          style="vertical-align:top;width:45px;box-sizing:border-box;height:20px;padding:0 2px;font-size:12px;"
          type="text"
          value="38"
        />
        <span
          style="display:inline-block;height:100%;font-size:13px;vertical-align:top;padding-left:4px;line-height:20px;"
        >
          %
        </span>
      </div>
    </div>
    <div
      data-color="1"
      style="margin-bottom:4px;height:19px;"
    >
      <div
        style="display:inline-block;position:relative;margin-bottom:4px;"
      >
        <span
          style="display:inline-block;height:100%;font-size:13px;vertical-align:top;width:20px;line-height:20px;"
        >
          Y:
        </span>
        <input
          style="vertical-align:top;width:45px;box-sizing:border-box;height:20px;padding:0 2px;font-size:12px;"
          type="text"
          value="0"
        />
        <span
          style="display:inline-block;height:100%;font-size:13px;vertical-align:top;padding-left:4px;line-height:20px;"
        >
          %
        </span>
      </div>
    </div>
    <div
      data-color="1"
      style="margin-bottom:4px;height:19px;"
    >
      <div
        style="display:inline-block;position:relative;margin-bottom:4px;"
      >
        <span
          style="display:inline-block;height:100%;font-size:13px;vertical-align:top;width:20px;line-height:20px;"
        >
          K:
        </span>
        <input
          style="vertical-align:top;width:45px;box-sizing:border-box;height:20px;padding:0 2px;font-size:12px;"
          type="text"
          value="5"
        />
        <span
          style="display:inline-block;height:100%;font-size:13px;vertical-align:top;padding-left:4px;line-height:20px;"
        >
          %
        </span>
      </div>
    </div>
  </div>
  <div
    data-color="1"
    style="position:absolute;left:318px;top:319px;"
  >
    <div
      style="display:inline-block;position:relative;margin-bottom:4px;"
    >
      <span
        style="display:inline-block;height:100%;font-size:13px;vertical-align:top;width:20px;line-height:20px;"
      >
        #
      </span>
      <input
        style="vertical-align:top;width:80px;box-sizing:border-box;height:20px;padding:0 2px;font-size:12px;"
        value="2196f3"
      />
      <span
        style="display:inline-block;height:100%;font-size:13px;vertical-align:top;padding-left:4px;line-height:20px;"
      />
    </div>
  </div>
</div>
`;
